### PR TITLE
fix(field): wikilink-aware inline-field parsing + stop migration deleting code blocks

### PR DESCRIPTION
**Pull request by chhoumann**
Two independent deepsec BUG findings, fixed at the root and proven in a live Obsidian vault. Kept as two commits because they are separate file areas.

---

## 1. HIGH (silent data loss) - migration deleted fenced-code-block content
`docs/static/scripts/migrateDataviewToFrontmatter.js`

The inline-field -> frontmatter migration scanned the note body line by line and treated **any** `key:: value` line as an inline field, with no awareness of fenced code blocks or inline code. Lines such as `Reference:: [[x]]` inside a ```` ```dataview ```` fence (or any code sample) were stripped from the body and pulled into frontmatter - silently destroying the user's code-block content. The global `\n{3,}` blank-line collapse could also delete blank lines inside code blocks and mangle frontmatter block scalars.

**Root fix** - `parseInlineFieldsWithWikilinks` is now code-aware:
- Track fenced code blocks (```` ``` ````/`~~~`) line by line with CommonMark-ish rules: 0-3 space indent, info string allowed on the opener, and a closer must match the marker char, be at least as long, and carry **no** info string (so ```` ```js ```` inside a ```` ``` ```` block does not falsely close it). CRLF closers handled. Every fence line and every line inside a fence is preserved verbatim and never migrated.
- **Blockquoted fenced blocks are tracked too** (see extended coverage below).
- Mask inline code spans (any backtick-run length) before field detection, and require the field-**name** region to be free of inline code - so a code span sitting before a real field never causes the whole line (and its code) to be deleted.
- Collapse blank lines only **outside** fences, and never touch the frontmatter.
- Strip the body via `vault.process` so the read + rewrite is atomic.

### Live red -> green (real script, real Obsidian app)
Ran the **actual** on-disk script via the Obsidian CLI `eval` against notes containing code blocks.

**Before (master):**
- default config: `Reference:: [[Inside Fence Should Survive]]` inside a ```` ```dataview ```` block was **deleted**.
- migrate-all: the **entire** body of a ```` ```markdown ```` block (`Author::`, `Status::`, blank lines, `BodyAfterBlankLines::`) was **deleted**.

**After:** `fencesPreservedVerbatim: true` for both - code blocks survive byte-for-byte, while real outside-code fields still migrate (with `[[Note, with comma]]` kept as one value).

### Extended coverage (update): blockquoted fences
A fenced code block can live inside a blockquote (`> ```lang` ... `> ...` ... `> ``` `, including nested `>>`). These are unambiguously code, so they are now tracked and preserved like top-level fences: `stripBlockquote()` peels blockquote markers, the opening quote depth is recorded, and the closer must sit at the same depth. If a line carries **fewer** markers, the blockquote container has ended - so the fence ends there and the line is reprocessed normally, which means a real field after a quoted block still migrates (no false-negative). Field detection still runs on the raw line, so blockquoted *plain* fields are unchanged.

Live red -> green (migrate-all, real script) on a note with a blockquoted ```` ```dataview ```` block + an outside field + a list-indented field:
- **Before (this PR's first revision):** `> Reference:: [[Inside Quoted Fence]]` inside the quoted fence was **deleted** (garbage-keyed into frontmatter as `"> Reference"`).
- **After:** the quoted-fence line is **preserved verbatim**; `Reference:: [[Real Outside]]` still migrates; and the list-indented `Status:: active under list` still migrates (no new false-negative).

### Decision: 4-space indented code blocks (out of scope, reasoned)
I assessed indented code blocks and deliberately did **not** implement them. Distinguishing a genuine indented code block from list-continuation text requires list/paragraph block context that a line-based script lacks:
- CommonMark: an indented code block cannot interrupt a paragraph and must follow a blank line / block boundary; a 4-space indent **under a list item is list content, not code**; loose lists even put a blank line before such continuation.
- The two lines are textually identical but must be classified oppositely: `␣␣␣␣Reference:: [[x]]` at body top after a blank line is indented code (should *not* migrate), but the same line under `- item` is list continuation (Dataview parses it as a field, so it *should* migrate).
- A partial heuristic would therefore either reintroduce data loss (deleting indented code) **or** cause the opposite, maintainer-flagged bug - silently skipping a real inline field merely indented under a list. Only indented code at the very start of the body is unambiguous, which is too narrow to justify a separate indented-block parser that would still leave the general case open and imply "indented code is handled" when it is not.

Residual (documented in code): an inline-field-like line inside a genuine 4-space *indented* code block may still be migrated. This is narrow (fenced blocks are the modern norm; dataview users use ```` ```dataview ```` fences) and is the safer side of the false-negative tradeoff. Fenced code (top-level **and** blockquoted) and inline code are fully preserved.

Per project convention, tests for this example userscript are **not** committed (validated locally via a `vm`-loaded harness, 50 assertions covering fences, blockquoted fences, container-exit, inline code, CRLF, frontmatter block scalars, and the list-indented false-negative guard, plus the live runs above).

---

## 2. Minor - garbled field-value autocomplete from wikilink-unaware comma split
`src/utils/InlineFieldParser.ts`, `src/utils/DataviewIntegration.ts`, new `src/utils/splitWikilinkAwareList.ts`

Inline field value parsing split on **every** comma, so a wikilink target containing a comma (`Related:: [[Note, with comma]]`) was shredded into `[[Note` and `with comma]]`. These feed the read-only field-value autocomplete (`FieldValueCollector`, `quickAddApi.fieldSuggestions`). No data-write or security impact.

**Root fix** - factor a single wikilink-aware splitter (`splitWikilinkAwareList`) that only splits on commas outside `[[ ... ]]` spans, and use it at all three duplicated split sites (`InlineFieldParser` once, `DataviewIntegration` twice). Plain comma lists stay byte-identical; wikilink commas are preserved. (The `DataviewIntegration` sites are mostly DRY + hardening, since Dataview pre-parses wikilinks into Link objects.)

### Live red -> green (real plugin API)
`fieldSuggestions.getFieldValues('Related', { includeInline: true })` against a note with `Related:: [[Note, with comma]]` and `[[Another, with comma]]`:
- **Before:** `["[[Another", "[[Note", "[[Plain Note]]", "with comma]]"]`
- **After:** `["[[Another, with comma]]", "[[Note, with comma]]", "[[Plain Note]]"]`
- Regression guard: `PlainTest:: alpha, beta, gamma` still returns `["alpha","beta","gamma"]`.

Committed unit tests: `splitWikilinkAwareList.test.ts` (helper) + wikilink cases in `InlineFieldParser.test.ts`.

---

## Validation
- Gates green: `tsc -noEmit`, `eslint .`, `svelte-check` (0 errors), `vitest` (3330 passed / 0 failed).
- Both bugs reproduced on current master and proven fixed in this worktree's isolated Obsidian vault.
- Adversarial review (opposing model) at design and on each diff, including the blockquote extension - design-stage high-severity findings folded in (notably the blockquote container-exit rule that prevents a mass missed-migration); final-diff reviews returned no real issues.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved inline field parsing so commas inside wikilinks, aliases, and embeds are preserved and split correctly.
  * Prevented inline-field migration from altering frontmatter and fenced code blocks (including quote/nesting handling).
  * Improved Dataview field value parsing to use wikilink-aware comma splitting for more accurate extraction.
* **Tests**
  * Added a new test suite for wikilink-aware list splitting.
  * Added inline-field parsing tests covering wikilink comma edge cases.

### CODE REVIEW

File: docs/static/scripts/migrateDataviewToFrontmatter.js [hunk 1]
```diff
@@ -225,26 +273,75 @@ function parseInlineFieldsWithWikilinks(content, allowedProperties = [], migrate
...
+    return { char: match[1][0], len: match[1].length };
+}
+
+/**
+ * A closing fence uses the same marker character, is at least as long as the
+ * opening fence, and carries no info string (only optional trailing whitespace).
+ */
+function isClosingFence(line, fenceChar, fenceLen) {
+    // Allow a trailing CR so CRLF files close their fences correctly.
+    const match = line.match(/^ {0,3}(`{3,}|~{3,})[ \t]*\r?$/);
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply docs/static/scripts/migrateDataviewToFrontmatter.js]: Thanks - half of this is now done, and the other half is a deliberate decision:

- **Blockquoted fences:** handled as of f9909e4, but via a `stripBlockquote()` depth model rather than the suggested `(?:> ?)*` regex. The regex would let a closer at a *different* quote depth (or a `> ```` that is literal code inside a top-level fence) close the wrong block; tracking the open quote depth and requiring the closer at the same depth (with a container-exit when a line drops markers) is more robust. Proven live: a `> ```dataview` block's content now survives byte-for-byte.

- **4-space indented code blocks:** deliberately *not* protected, and the suggested `/^(?: {4,}|\t)/ → preserve` heuristic is exactly what I avoided. It reintroduces the opposite bug: a real inline field indented under a list item (`- item` / `    status:: active`) is valid list content that Dataview parses as a field and users expect to migrate - the heuristic would silently skip it (a missed migration). Distinguishing genuine indented code from list/paragraph continuation needs block context a line-based script doesn't have (CommonMark: indented code can't interrupt a paragraph; 4-space under a list is list content; loose lists even put a blank line first). This is documented in the code and the PR as a narrow, reasoned residual; fenced (top-level + blockquoted) and inline code are fully preserved.
